src/nvim/shada.c

```diff
@@ -1153,8 +1153,8 @@ static void shada_read(ShaDaReadDef *const sd_reader, const int flags)
   }
   HistoryMergerState hms[HIST_COUNT];
   if (srni_flags & kSDReadHistory) {
-    for (uint8_t i = 0; i < HIST_COUNT; i++) {
-      hms_init(&hms[i], i, (size_t)p_hi, true, true);
+    for (int i = 0; i < HIST_COUNT; i++) {
+      hms_init(&hms[i], (uint8_t)i, (size_t)p_hi, true, true);
```

Comment:
I think there might be a slightly bigger issue here:

`HIST_COUNT` is an enum `HistoryType` which starts at index `-2`, but then we have `hms_init()` which assumes that `const uint8_t history_type`
```c
static inline void hms_init(HistoryMergerState *const hms_p, const uint8_t history_type,
                            const size_t num_elements, const bool do_merge, const bool reading)
```
we can't simply swap the argument in this function either because `HistoryMergerState` also uses `uint8_t` for history type.

Either way, I think it's better to use the actual type for consistency
```suggestion
    for (HistoryType i = 0; i < HIST_COUNT; i++) {
      hms_init(&hms[i], (uint8_t)i, (size_t)p_hi, true, true);

```